packages/kv-asset-handler/test/serveSinglePageApp.test.ts

```diff
@@ -1,44 +1,46 @@
-import test from "ava";
-import { serveSinglePageApp } from "../index";
-import { mockGlobalScope, mockRequestScope } from "../mocks";
+import { beforeEach, expect, test } from "vitest";
+import { serveSinglePageApp } from "../src/index";
+import { mockGlobalScope, mockRequestScope } from "./mocks";
 
-mockGlobalScope();
+beforeEach(() => {
+	mockGlobalScope();
+	mockRequestScope();
+});
 
 function testRequest(path: string) {
-	mockRequestScope();
 	const url = new URL("https://example.com");
 	url.pathname = path;
 	const request = new Request(url.toString());
 
 	return request;
 }
 
-test("serveSinglePageApp returns root asset path when request path ends in .html", async (t) => {
+test("serveSinglePageApp returns root asset path when request path ends in .html", async () => {
 	const path = "/foo/thing.html";
 	const request = testRequest(path);
 
 	const expected_request = testRequest("/index.html");
 	const actual_request = serveSinglePageApp(request);
 
-	t.deepEqual(expected_request, actual_request);
+	expect(expected_request).toEqual(actual_request);
 });
 
-test("serveSinglePageApp returns root asset path when request path does not have extension", async (t) => {
+test("serveSinglePageApp returns root asset path when request path does not have extension", async () => {
 	const path = "/foo/thing";
 	const request = testRequest(path);
 
 	const expected_request = testRequest("/index.html");
 	const actual_request = serveSinglePageApp(request);
 
-	t.deepEqual(expected_request, actual_request);
+	expect(expected_request).toEqual(actual_request);
```

Comment:
Can you compare `Request` objects this way with vitest?

In a quick test in a new `npm create cloudflare@latest` project, this passes:

```ts
	it('test', async () => {
		const request1 = new Request('https://example.com/foo');
		const request2 = new Request('https://example.com/bar');

		expect(request1).toEqual(request2);
	});
```

Checking `req.url` matches is probably sufficient for these SPA tests?

Reply:
Gosh! Let me look into that! Thanks for keeping us honest @Cherry

Reply:
https://github.com/cloudflare/workers-sdk/pull/11421